python/mxnet/ndarray/ndarray.py

```diff
@@ -1053,17 +1053,15 @@ def _advanced_index_to_array(idx, ax_len, ctx):
             if idx.dtype != idx_dtype:
                 idx = idx.astype(idx_dtype)
             return idx.as_in_context(ctx)
-
         elif isinstance(idx, (np.ndarray, list, tuple)):
             return array(idx, ctx, idx_dtype)
-
         elif isinstance(idx, integer_types):
             return array([idx], ctx, idx_dtype)
-
         elif isinstance(idx, py_slice):
             start, stop, step = idx.indices(ax_len)
             return arange(start, stop, step, ctx=ctx, dtype=idx_dtype)
-
+        elif sys.version_info[0] > 2 and isinstance(idx, range):
```

Comment:
Shall we also update the doc?

Reply:
Good point. Will send in the doc in a separate PR.